### PR TITLE
feat(web_console): raw data form

### PR DESCRIPTION
**Pull request by AlllenShen**
* fix: env not displayed in federation

### CODE REVIEW

File: web_console/components/Form.jsx [hunk 1]
```diff
@@ -40,28 +40,60 @@ export default function Form({
...
-  const [form, setForm] = useState(fields.reduce((total, current) => {
-    total[current.key] = current.hasOwnProperty('value')
-      ? current.value
-      : current.value || current.default;
-    return total;
-  }, {}));
+
+  const mapFields2Form = fields => {
+    // flat all group fileds
+    fields = fields.reduce((total, curr) => {
```
Comment (marswong) [inline web_console/components/Form.jsx:50]: why update `fields` directly?

File: web_console/components/Form.jsx [hunk 2]
```diff
@@ -40,28 +40,60 @@ export default function Form({
...
+    const formData = fields.reduce((total, current) => {
+      total[current.key] = current.hasOwnProperty('value')
+        ? current.value
+        : current.value || current.default;
+      return total;
+    }, {})
+    return [fields, formData]
+  }
+  let formData
+  [fields, formData] = mapFields2Form(fields)
```
Comment (marswong) [inline web_console/components/Form.jsx:71]: the same issue

File: web_console/pages/admin/federation.jsx [hunk 3]
```diff
@@ -12,50 +12,8 @@ import { createFederation, updateFederation } from '../../services';
...
-    return data[path[0]]
-  }
-  const currPathIsArray = path[0].endsWith('[]')
-  let currPath = path[0].replace('[]', '')
-  let nextLayer = currPathIsArray ? data[currPath][0] : data[path[0]]
-  if (nextLayer === undefined) return undefined
-  return getValueFromJson(nextLayer, path.slice(1))
-}
+import { fillJSON, getValueFromJson, getParsedValueFromData } from '../../utils/form_utils'
+import { loadAll } from 'js-yaml';
```
Comment (marswong) [inline web_console/pages/admin/federation.jsx:16]: dirty import

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "requires": true,
+  "lockfileVersion": 1,
+  "dependencies": {
+    "esm": {
```
Comment (marswong) [inline package-lock.json:5]: if `esm` is needed, please add to `devDependencies` and regenerate `package-lock.json`, and remember to lock version

**Review by marswong — CHANGES_REQUESTED** [review]

File: web_console/components/Form.jsx (hunk 1 above)
Comment (AlllenShen) [reply web_console/components/Form.jsx:50]: cause `fields` are used in existing code, update it to feet them. It shouldn't be implemented like this, i'll update later

**Review by marswong — APPROVED** [review]

**Pull request merged**